### ISSUE TITLE
Upgrading from v0.18.1 to  v0.19.0 errors with `failed to start child: Wallaby.Experimental.Chrome.Chromedriver` (macOS only, no issues with Linux)

### ISSUE THREAD

**Issue by rickclare** [keathley/wallaby#290]
After upgrading Wallaby from **v0.18.1** to **v0.19.0** our test suite no longer starts, **but only on macOS**.

I'm using the following setup:
* macOS Sierra v10.12.6
* Google Chrome v60.0.3112.90
* chromedriver v2.31.488774 
* Erlang OTP 20.0
* Elixir 1.5.1

_Note: We are *not* encountering this issue when running on Linux (test suite runs successfully with 	Ubuntu 16.04.3 LTS in a docker container, using Wallaby v0.19.0)_

When starting our test suite with `mix test` we immediately receive the following error message:

```
INFO REPORT==== 10-Aug-2017::12:10:51 ===
    application: logger
    exited: stopped
    type: temporary
** (Mix) Could not start application wallaby: Wallaby.start(:normal, []) returned an error: shutdown: failed to start child: Wallaby.Experimental.Chrome
    ** (EXIT) shutdown: failed to start child: Wallaby.Experimental.Chrome.Chromedriver
        ** (EXIT) an exception was raised:
            ** (ErlangError) Erlang error: :eacces
                :erlang.open_port({:spawn_executable, "/Volumes/MBP_Data/Dev/my_app/_build/test/lib/wallaby/priv/run_command.sh"}, [:binary, :stream, :use_stdio, :exit_status, {:args, ["/usr/local/bin/chromedriver", "--port=64289"]}])
                (wallaby) lib/wallaby/experimental/chrome/chromedriver.ex:19: Wallaby.Experimental.Chrome.Chromedriver.init/1
                (stdlib) gen_server.erl:365: :gen_server.init_it/2
                (stdlib) gen_server.erl:333: :gen_server.init_it/6
                (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
```

Relevant config from our `config/test.exs`: _(not changed since upgrade from v0.18.1 to v0.19.0)_

```elixir
...
config :wallaby,
  max_wait_time: 5000,
  driver: Wallaby.Experimental.Chrome,
  chrome: [
    headless: true
  ]
...
```

Contents of our `test/test_helper.exs`: _(not changed since upgrade from v0.18.1 to v0.19.0)_

```elixir
Faker.start

Application.put_env :wallaby, :base_url, MyAppWeb.Endpoint.url
{:ok, _} = Application.ensure_all_started :ex_machina

{:ok, _} = Application.ensure_all_started :wallaby

formatters =
  if System.get_env("CIRCLECI") || System.get_env("CI") || System.get_env("TRAVIS"),
    do: [JUnitFormatter, ExUnit.CLIFormatter],
  else: [ExUnit.CLIFormatter]

ExUnit.configure formatters: formatters

ExUnit.start()

Ecto.Adapters.SQL.Sandbox.mode(MyApp.Repo, :manual)
```

This may well be to do with my installation of Chrome or chromedriver, but I'm unsure how to debug further.  Is is possible to configure Wallaby to enable verbose chromedriver logging to stdout or a chromedriver.log log-file, to see if chromedriver is reporting more info?

**Comment by keathley** [comment]
Thanks for the detailed report @rickclare. I'm trying to reproduce this locally but so far I haven't had any luck. I'm running the same versions as you are so I'm not sure whats up yet. Did you install chromedriver through brew?

**Comment by keathley** [comment]
@rickclare finally managed to figure something out. Looks like when wallaby is downloaded the script that we wrap chromedriver in doesn't have execute permissions. I'm guessing something either went wrong when we published to hex or hex has changed the way it handles these files. I'll try to follow up tomorrow to see whats going on. The fact that its still working on linux seems to indicate that it might be a bug in hex. I'll try to follow up tomorrow to see if I can track down the root cause.

**Comment by rickclare** [comment]
That's great, thanks for your work on this @keathley. Let me know if I can help with debugging further

**Comment by rickclare** [comment]
@keathley I can confirm that updating wallaby to *v0.19.0* and executing the following resolves that issue locally.

```shell
mix deps.update wallaby # update to *v0.19.0*
chmod a+x _build/test/lib/wallaby/priv/run_command.sh
mix test
```

Well spotted, that must have been a pain to find!  Hope you find a fix that gets around the potential _hex_ problem. Very strange that this worked for me in v0.18.0 and v0.18.1, I wonder what has changed in hex?

**Comment by keathley** [comment]
This is the related issue on hex: https://github.com/hexpm/hex/issues/404. Looks like its been fixed. I'll try to push a new version and make sure things are looking OK.

**Comment by rickclare** [comment]
Hi @keathley 

Unfortunetely v0.19.1 _hasn't fixed the issue:

``` shell
$ grep vsn _build/test/lib/wallaby/ebin/wallaby.app                                       
{vsn,"0.19.1"},

$ls -l _build/test/lib/wallaby/priv/                                                     
total 16
-rw-r--r--  1 rick  admin  124 13 Aug 16:12 run_command.sh
-rw-r--r--  1 rick  admin   83 13 Aug 16:12 run_phantom.sh
```

**Comment by keathley** [comment]
@rickclare can you rm -rf your deps and try again? Should be fixed now.

**Comment by rickclare** [comment]
@keathley Yes, a `rm -rf deps` fixes it.   Excellent, thanks for your help and persistence on this.  Should I close this issue, or would you prefer to close it?

**Comment by keathley** [comment]
No problem. I had accidentally published 0.19.1 with the incorrect hex version. Had to do a quick cleanup and re-publish before I missed my window. I can close it for now. If something else crops up feel free to re-open :+1:.
